manifests/repomd_gpg.pp

```diff
@@ -0,0 +1,64 @@
+# == Class: certs
+# Sets up the GPG key for Pulp repository metadata signing
+class certs::repomd_gpg (
+  $enable                  = $::certs::enable_repomd_gpg,
+  $regenerate              = $::certs::regenerate_repomd_gpg,
+  $dir                     = $::certs::repomd_gpg_dir,
+  $user                    = $::certs::repomd_gpg_user,
+  $group                   = $::certs::repomd_gpg_group,
+  $gpg_name                = $::certs::repomd_gpg_name,
+  $comment                 = $::certs::repomd_gpg_comment,
+  $email                   = $::certs::repomd_gpg_email,
+) {
+  if $enable {
+
+    $gpg_key_params = "Key-Type: RSA\nKey-Length:4096\nExpire-Date: 0\n"
+    $gpg_name_param = $gpg_name ? { undef => '', default => "Name-Real: ${gpg_name}\n", }
+    $gpg_comment_param = $comment ? { undef => '', default => "Name-Comment: ${comment}\n", }
+    $gpg_email_param = $email ? { undef => '', default => "Name-Email: ${email}\n", }
+    $gpg_name_params = "${gpg_name_param}${gpg_comment_param}${gpg_email_param}"
+    $gpg_params = "${gpg_key_params}${gpg_name_params}%no-protection\n%commit\n"
+    if $regenerate {
+      file { $dir:
```

Comment:
This will lead to a duplicated resource below. I think the only way is an exec. Also ensure it's executed before the directory is created again.